metadata_writer_sqlite_monitor.c

```diff
@@ -49,7 +49,20 @@ static uint8_t md_sqlite_monitor_dump_json(struct md_writer_sqlite *mws, FILE *o
 {
     sqlite3_reset(mws->dump_monitor);
 
-    return md_json_helpers_dump_write(mws->dump_monitor, output);
+    json_object *jarray = json_object_new_array();
+
+    if (md_json_helpers_dump_write(mws->dump_monitor, jarray) ||
+        json_object_array_length(jarray) == 0)
+    {
+        json_object_put(jarray);
+        return RETVAL_FAILURE;
+    }
+
+    const char *json_str = json_object_to_json_string_ext(jarray, JSON_C_TO_STRING_PLAIN);
+    fprintf(output, "%s", json_str);
+
+    json_object_put(jarray);
+    return RETVAL_SUCCESS;
 }
 
```

Comment:
I just added a comment here because I seemingly can comment on the whole commit :)

All the different `..._dump_db_json()` functions seem to do exactly the same, with the exception of the callback passed to `dump_write`. Perhaps the shared logic can be merged in one helper function?